prod/prod.go

```diff
@@ -18,43 +18,109 @@ package prod
+
+// ErrPublisher is returned when the publisher string cannot be decoded into bytes
+var ErrPublisher = errors.New("failed to decode publisher")
+
+// ErrPubKey is returned when the publisher bytes cannot be decompressed as a public key
+var ErrPubKey = errors.New("failed to decompress public key")
+
+// ErrFeedLookup is returned when the recovery feed cannot be successefully looked up
+var ErrFeedLookup = errors.New("failed to look up recovery feed")
+
+// ErrFeedContent is returned when there is a failure to retrieve content from the recovery feed
+var ErrFeedContent = errors.New("failed to get content for recovery feed")
+
+// ErrTargets is returned when there is a failure to unmarshal the feed content as a trojan.Targets variable
+var ErrTargets = errors.New("failed to unmarshal targets in recovery feed content")
+
+// RecoveryHook defines code to be executed upon failing to retrieve pinned chunks
 type RecoveryHook func(ctx context.Context, chunkAddress chunk.Address) error
 
 // sender is the function call for sending trojan chunks
 type sender func(ctx context.Context, targets trojan.Targets, topic trojan.Topic, payload []byte) (*pss.Monitor, error)
 
 // NewRecoveryHook returns a new RecoveryHook with the sender function defined
-func NewRecoveryHook(send sender) RecoveryHook {
+func NewRecoveryHook(send sender, handler feed.GenericHandler) RecoveryHook {
 	return func(ctx context.Context, chunkAddress chunk.Address) error {
-		targets, err := getPinners(chunkAddress)
+		targets, err := getPinners(ctx, handler)
 		if err != nil {
 			return err
 		}
 		payload := chunkAddress
-		topic := trojan.NewTopic("RECOVERY")
 
-		// TODO: monitor return should
-		if _, err := send(ctx, targets, topic, payload); err != nil {
+		// TODO: returned monitor should be made use of
+		if _, err := send(ctx, targets, RecoveryTopic, payload); err != nil {
 			return err
 		}
 		return nil
 	}
 }
 
-// TODO: refactor this method to implement feed of target pinners
-// getPinners returns the specific target pinners for a corresponding chunk address
-func getPinners(chunkAddress chunk.Address) (trojan.Targets, error) {
+// getPinners returns the specific target pinners for a corresponding chunk
+func getPinners(ctx context.Context, handler feed.GenericHandler) (trojan.Targets, error) {
+	// TODO: obtain fallback Publisher from cli flag if no Publisher found in Manifest
+	// get feed user from publisher
+	publisher, _ := ctx.Value("publisher").(string)
+	publisherBytes, err := hex.DecodeString(publisher)
+	if err != nil {
+		return nil, ErrPublisher
+	}
+	pubKey, err := crypto.DecompressPubkey(publisherBytes)
+	if err != nil {
+		return nil, ErrPubKey
+	}
+	addr := crypto.PubkeyToAddress(*pubKey)
+
+	// get feed topic from trojan recovery topic
+	topic, err := feed.NewTopic(RecoveryTopicText, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// read feed
+	fd := feed.Feed{
+		Topic: topic,
+		User:  addr,
+	}
+	query := feed.NewQueryLatest(&fd, lookup.NoClue)
```

Comment:
is this always `lookup.NoClue`? does it not store latest. state to be. used as hint?

Reply:
as of now, we do not have any idea when the last update to the feed happened, so this is why `NoClue` is used.

but i have added a note to #2181, we can further optimize it this way i think.

thanks